File: Fortran/testOnMult.f90
subroutine OnMult(m_ar, m_br)
    integer, intent(in) :: m_ar, m_br
    integer :: Time1, Time2, clock_rate, clock_max
    real*8, allocatable :: pha(:,:), phb(:,:), phc(:,:)
    real*8 :: temp, timing
    integer :: i, j, k, size
    integer :: unit

    allocate(pha(m_ar, m_ar), phb(m_ar, m_ar), phc(m_ar, m_ar))

    do i = 1, m_ar
        do j = 1, m_ar
            pha(i, j) = 1.0d0
        end do
    end do

    do i = 1, m_br
        do j = 1, m_br
            phb(i, j) = dble(i)
        end do
    end do

    call SYSTEM_CLOCK(count=Time1, count_rate=clock_rate, count_max=clock_max)

    do i = 1, m_ar
        do j = 1, m_br
            temp = 0.0d0
            do k = 1, m_ar
                temp = temp + pha(i, k) * phb(k, j)
            end do
            phc(i, j) = temp
        end do
    end do
    call SYSTEM_CLOCK(count=Time2)

    timing = real(Time2 - Time1) / real(clock_rate)

    ! Open CSV file for appending
    open(unit=unit, file='single_mult_fortran.csv', status='old', action='write', position='append', iostat=ierr)
    if (ierr /= 0) then
        print *, "Error opening file!"
        stop
    end if

    ! Write size and timing to CSV file
    if (unit == 10) then
        write(unit, '(A)') "size, time"
    end if
    write(unit, '(I0, A, F0.3)') m_ar, ",", timing

    ! Close CSV file
    close(unit)

    deallocate(pha, phb, phc)
end subroutine OnMult


program Main
    implicit none
    integer :: size, m_ar, m_br
    real*8 :: timing

    ! Initialize starting size
    size = 600

    ! Loop until size reaches 3000
    do while (size <= 3000)
        ! Set matrix dimensions
        m_ar = size
        m_br = size

        call OnMult(m_ar, m_br)

        call OnMult(m_ar, m_br)

        ! Increment size by 400
        size = size + 400
    end do

end program Main

File: java/src/TestOnMult.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Locale;

public class TestOnMult {

    public static void onMult(int m_ar, int m_br, FileWriter writer) throws IOException {
        long Time1, Time2;
        String st;
        double temp;
        int i, j, k;

        double[] pha, phb, phc;

        pha = new double[m_ar * m_ar];
        phb = new double[m_ar * m_ar];
        phc = new double[m_ar * m_ar];

        for (i = 0; i < m_ar; i++)
            for (j = 0; j < m_ar; j++)
                pha[i * m_ar + j] = 1.0;

        for (i = 0; i < m_br; i++)
            for (j = 0; j < m_br; j++)
                phb[i * m_br + j] = i + 1;

        Time1 = System.currentTimeMillis();

        for (i = 0; i < m_ar; i++) {
            for (j = 0; j < m_br; j++) {
                temp = 0;
                for (k = 0; k < m_ar; k++) {
                    temp += pha[i * m_ar + k] * phb[k * m_br + j];
                }
                phc[i * m_ar + j] = temp;
            }
        }

        Time2 = System.currentTimeMillis();
        double t = (double) (Time2 - Time1) / 1000;
        st = String.format(Locale.US, "%s,%.3f\n", m_ar, t);
        writer.write(st);
        writer.flush();
    }

    public static void main(String[] args) {
        FileWriter writer = null;
        try {
            File file = new File("single_mult_java.csv");
            writer = new FileWriter(file);
            writer.write("size,time\n"); // Column headers
            for (int i = 600; i <= 3000; i += 400) {
                for (int j = 0; j < 2; j++) {
                    onMult(i, i, writer);
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (writer != null) {
                try {
                    writer.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}